### ISSUE TITLE
Support appending files with `Bun.write`

### ISSUE THREAD

**Issue by lsyuanyuan** [oven-sh/bun#10473]
### What is the problem this feature would solve?

 I read the document and the current way `Bun` writes files is:

```js
const data = `It was the best of times, it was the worst of times.`;
await Bun.write("output.txt", data);
```

If new content needs to be added and written, then the `node: fs` module needs to be used,

```js
import { appendFile } from "node:fs/promises";

await appendFile("message.txt", "data to append");
```

Is it being considered to add an optional parameter to Bun.write() or introduce a new method like Bun.appendWrite() to indicate appending new content? I feel this aligns well with Bun's inherent characteristics and makes it more concise

### What is the feature you are proposing to solve the problem?

Add an optional parameter to `Bun.write()` or introduce a similar method `Bun.appendWrite()`

### What alternatives have you considered?

Using the solution introduced in the official documentation

**Comment by mangs** [comment]
`FileSink` does what you're looking for: https://bun.sh/docs/api/file-io#incremental-writing-with-filesink

**Comment by gutenye** [comment]
FileSink does not, it clears file content first before writing.

**Comment by mangs** [comment]
You're right, I realized that after I wrote this and forgot to come back and change this. Thank you for making this clear.

**Comment by VesperQuartz** [comment]
Yes we need this!

**Comment by KilianB** [comment]
Appending to a file is such a common occurrence that at the end of the day I always have to fall back to the fs module.

**Comment by Atulin** [comment]
Crazy that appending to a file is not a thing yet

**Comment by makhlouf1102** [comment]
it would be nice !

**Comment by MrDanielHarka** [comment]
I need this!
